File: zettelkasten-front/src/models/Stats.ts
export interface DailyStats {
  date: Date;
  cards_created: number;
  tasks_created: number;
  tasks_completed: number;
}

export interface DailyStatsResponse {
  stats: DailyStats[];
  total: {
    cards_created: number;
    tasks_created: number;
    tasks_completed: number;
  };
}

File: go-backend/models/stats.go
package models

import "time"

// DailyStats represents activity statistics for a single day
type DailyStats struct {
	Date           time.Time `json:"date"`
	CardsCreated   int       `json:"cards_created"`
	TasksCreated   int       `json:"tasks_created"`
	TasksCompleted int       `json:"tasks_completed"`
}

// DailyStatsResponse is the API response for daily stats
type DailyStatsResponse struct {
	Stats []DailyStats `json:"stats"`
	Total struct {
		CardsCreated   int `json:"cards_created"`
		TasksCreated   int `json:"tasks_created"`
		TasksCompleted int `json:"tasks_completed"`
	} `json:"total"`
}
